aesara/tensor/math_opt.py

```diff
@@ -1862,6 +1862,71 @@ def local_neg_div_neg(fgraph, node):
                     return [true_div(new_num, denom)]
 
 
+@register_canonicalize
```

Comment:
Is canonicalization applied only once? In which case, should we also register these for the stabilize and specialize phases so that the rewrites are used in intermediate graphs as well?

Reply:
> should we also register these for the stabilize and specialize phases

No, if those phases need canonicalization, they should apply it themselves in its entirety, or apply specific canonicalization steps themselves&mdash;i.e. canonicalization steps should not register to other optimizers or DBs.